src/mds/MDSMap.cc

```diff
@@ -610,37 +610,9 @@ void MDSMap::encode(bufferlist& bl, uint64_t features) const
   }
 
   using ceph::encode;
-  if ((features & CEPH_FEATURE_PGID64) == 0) {
```

Comment:
should only remove code block  if ((features & CEPH_FEATURE_PGID64) == 0) and if ((features & CEPH_FEATURE_MDSENC) == 0), don't change anything else